### ISSUE TITLE
Remove the need to store the `__inputs` property on Outputs

### ISSUE THREAD

**Issue by ellismg** [pulumi/pulumi-kubernetes#566]
@hausdorff believes this is no longer needed.  We should remove it if possible, as it interacts poorly with secret inputs.

If we do need to keep it around, we should make changes in the provider to ensure secretness is copied into this copy of the inputs bag.

**Comment by hausdorff** [comment]
@ellismg I think I talked to you about this before, but here is a detailed breakdown of this issue just in case.

* It used to be the case that Kuberentes resources in the statefile (_i.e._, in `.deployment.resources`) took the form `{ inputs: (...), live: (...) }`. "Packing" the inputs and outputs into what should _only_ be the outputs was necessary to allow Kubernetes to perform an update, which involves a three-way merge between last inputs, current inputs, and current live version.
* We moved to the current `__inputs` model because of #137 -- basically output properties won't resolve unless the properties exist at the root of the resource state.

I mentioned this before, but the "right way" to fix this will basically involve changing the protocol to have last inputs built into the correct RPC calls.

I have a patch nearly ready, but it won't be checked in tomorrow -- it will probably go in within the next few days.

**Comment by ellismg** [comment]
> I mentioned this before, but the "right way" to fix this will basically involve changing the protocol to have last inputs built into the correct RPC calls.

Makes sense.

**Comment by pgavlin** [comment]
We also fill in the last-applied-configuration tag these days, and could use that instead.

**Comment by ellismg** [comment]
My fear is that the changes here are going to be too disruptive to do as we push towards the Q3 release.  I think that there are interesting forwards and backwards compatibility issues we'd need to address and it would also require changes to the RPC interfaces.  We should consider this for a future release.

**Comment by lblackstone** [comment]
The engine-side change is done, so this work is unblocked now. To maintain backwards compatibility with older versions of the engine, the provider will need to store `__inputs` until we decide to enforce a minimum engine version.
